lib/elixir/lib/integer.ex

```diff
@@ -400,17 +400,23 @@ defmodule Integer do
       iex> Integer.gcd(10, 0)
       10
 
+      iex> Integer.gcd(7, 7)
+      7
+
       iex> Integer.gcd(0, 0)
       0
 
   """
+  @spec gcd(0, 0) :: 0
+  @spec gcd(int, int) :: int when int: pos_integer
```

Comment:
When you say:

    fun(a, a) when a: type

You are saying that `a` is a subtype of type. That's all to it. You are not saying both have the same value, as you cannot affirm anything about the value of `a` unless `type` is a "value type".